### ISSUE TITLE
No async data on root index.html after getting error

### ISSUE THREAD

**Issue by antoineneff** [sveltejs/sapper#450]
If the root index.html file (`routes/index.html`) has some async data in its preload method (returning a promise or using async/await), the page won't reload after getting an error from `this.error`, the error layout is still showing up.

I'm getting 3 errors in the browser console when trying going back to `/` route :
```
TypeError: "each_value is undefined"
TypeError: this.proxyTarget is undefined (x2)
```

To reproduce, you can copy the `routes/blog/index.html` from sapper template to `routes/index.html`, get an error navigating to an invalid route and trying to go back on home from navigation. I did that on that repo : https://github.com/antoineneff/sapper-error-repro

Feel free to ask if you need further information.

**Comment by sgametrio** [comment]
Same bug here.
I'll link [here](https://github.com/sveltejs/sapper-template/issues/94) issue I opened on sapper-template too

**Comment by Conduitry** [comment]
Should be fixed by #620.
